common/status.py

```diff
@@ -0,0 +1,331 @@
+# SPDX-FileCopyrightText: © 2025 Samuel Moore
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of the program "Back In Time" which is released under GNU
+# General Public License v2 (GPLv2). See LICENSES directory or go to
+# <https://spdx.org/licenses/GPL-2.0-or-later.html>.
+"""Provide tools for managing and reporting status of backup profiles.
+
+Supports command-line arguments to filter output by profile, show only issues,
+or return the status in JSON format.
+
+Status file is stored at $XDG_STATE_HOME/backupstatus.json (typically:
+~/.local/state/backupstatus.json).
+"""
+import json
+import os
+import pathlib
+from pathlib import Path
+from datetime import datetime
+import argparse
+import config
+import logger
+import sshtools
+from exceptions import MountException
+import snapshots
+
+
+class BackupStatus:
+    """Context manager tracking and providng most recent backup status.
+
+    The status includes information about the last successful backup, the last
+    run completed, snapshot mode, and relevant paths. It can be retrieved in
+    either a human-readable or JSON-formatted string.
+
+    Args:
+        args (argparse.Namespace, optional):
+                        Command line arguments when called from command line.
+
+        cfg (config.Config):
+                        Current configuration object.
+
+    Command line options (via args):
+        --profile <name> or --profile-id <id>:
+                        Get the status of a specific profile.
+        --issues:
+                        Get the status of profiles with an unsuccessful
+                        last run.
+        --json:
+                        Get the status as a JSON-formatted string.
+
+
+    Returns:
+        `__repr__`:
+                returns a JSON string representation.
+        `__str__`:
+                returns a human-readable formatted string.
+    """
+    def __init__(self,
+                 args: argparse.Namespace = None,
+                 cfg: config.Config = None):
```

Comment:
I would prefer to decouple the class from `argparse` module.
My quick n dirty suggestion:

```suggestion
    def __init__(self,
                 cfg: config.Config = None,
                 profile: str = None,
                 profile_id: int = None,
                 with_issues: bool,  #  default value will come from the args object
                 as_json: bool):
```